### ISSUE TITLE
About object attitude and calculated attitude

### ISSUE THREAD

**Issue by panwenfeics26** [cviviers/YOLOv5-6D-Pose#26]
Hello, author! Now I am a little confused about the attitude estimated by the object. The attitude estimated by me on my data set is very abnormal. Can you tell me how the coordinate system of the object is determined? Is its origin the center of mass of the object? Below is my estimated attitude data

Here is the code to calculate the pose：
import argparse  #
import json
import os
from pathlib import Path
from threading import Thread
from datetime import datetime
import glob
import numpy as np
import torch
import yaml
from tqdm import tqdm
from  torch.cuda.amp import autocast
from models.experimental import attempt_load
from utils.datasets import create_dataloader
from utils.general import coco80_to_coco91_class, check_dataset, check_file, check_img_size, check_requirements, scale_coords, set_logging, increment_path, colorstr, retrieve_image
from utils.plots import plot_images, output_to_target, plot_study_txt
from utils.torch_utils import select_device, time_synchronized
from utils.pose_utils import box_filter, get_3D_corners, pnp, epnp, calcAngularDistance, compute_projection, compute_transformation, get_camera_intrinsic, fix_corner_order, calc_pts_diameter, MeshPly
from utils.loss import PoseLoss
import matplotlib.pyplot as plt
from PIL import Image
import matplotlib.patches as patches

from utils.compute_overlap import wrapper_c_min_distances # for computing ADD-S metric
#

def read_file(input_folder):
    data_pr=[]
    with open(input_folder, 'r') as f:#
        lines = f.readlines()
        for line in lines:
            parts=line.strip().split()[0:18]
            for i in range(len(parts)):
                number_GT=float(parts[i])
                data_pr.append(number_GT)

    return data_pr

def Calculation_juzhen(box_pr,corners3D,vertices,internal_calibration):#
    num_keypoints=9
    corners2D_pr = np.array(np.reshape(box_pr[:num_keypoints * 2], [num_keypoints, 2]), dtype='float32')
    R_pr, t_pr = pnp(np.array(np.transpose(np.concatenate((np.zeros((3, 1)), corners3D[:3, :]), axis=1)), dtype='float32'),corners2D_pr, np.array(internal_calibration, dtype='float32'))

    return R_pr,t_pr

def jisuan(detect_path1,corners3D,vertices,internal_calibration): #
    data_pr=read_file(detect_path1)
    R_pr,t_pr= Calculation_juzhen(data_pr,corners3D,vertices,internal_calibration)
    return R_pr, t_pr

if __name__ == "__main__":
    detect_path = 'shijishiyan/labels/000000.txt'  #
    #box_gt,box_pr=read_file(input_folder, input_folder2)

    mesh = MeshPly('../data/LINEMOD/aoliao2_new/registeredScene.ply')  #
    vertices = np.c_[np.array(mesh.vertices), np.ones((len(mesh.vertices), 1))].transpose()
    corners3D = get_3D_corners(vertices)
    u0=64.898
    v0=361.929
    fx=641.627
    fy=640.349
    internal_calibration = get_camera_intrinsic(u0, v0, fx, fy)
    R_pr,t_pr,=jisuan(detect_path,corners3D,vertices,internal_calibration)
    print(R_pr)
    print(t_pr)

[[    0.34614    0.018251      -0.938]
 [   -0.45972    -0.86825    -0.18654]
 [   -0.81783     0.49579    -0.29215]]
[[    -1.0103]
 [    0.78639]
 [    -3.1827]] 
And you can see that the estimate for the distance is -3.1827 which is obviously an anomaly
I hope you can answer that. Thank you very much！

**Comment by cviviers** [comment]
Hi @panwenfeics26,

Does the keypoint predictions on the image make sense? Can you show an example? If they are indeed correct, did you also confirm that your registeredScene.ply is at the correct scale? All the coordinates should also be in meters.
